APE17.rst

```diff
@@ -0,0 +1,701 @@
+affiliated and other packages using astropy-helpers change over to the new
+infrastructure.
+
+Once any issues have been ironed out in the core and coordinated packages, the
+migration guide should be advertised widely to all maintainers of packages
+using astropy-helpers. Throughout this process, astropy-helpers should continue
+to be supported, and once the migration guide is advertised the final date for
+support of astropy-helpers should be made clear. The recommendation in this
+APE is to support astropy-helpers until the end of the 4.0 LTS cycle at the
+latest (end of 2021), but this can be re-assessed depending on the ease of
+migration and the feasibility of maintaining astropy-helpers if there are
+extensive breaking changes in its dependencies.
+
+Impact for users
+----------------
+
+If done properly, these changes should have no noticeable impact for users.
+Users will still be able to pip install (or conda install when available)
+packages and for large packages such as the core astropy package, which contains
+a lot of compiled extensions, most users will not see any
+difference since they will be installing astropy from pre-built packages (wheels
+or conda packages). Running tests using ``package.test()`` will still be
+supported.
+
+Impact for non-core developers
+------------------------------
+
+The immediate impact for developers of packages that use astropy-helpers is
+having to update the layout and infrastructure in their packages to follow the
+new guidelines presented here. However, we emphasise that these changes are
+recommendations and not mandatory, and astropy-helpers will continue to work as
+expected as long as it is included as a submodule. However, we recommend that
+astropy-helpers no longer be actively developed, in which case guarantees could
+not be made that some aspects of the astropy-helpers infrastructure will not
+break with future releases of Python, setuptools, or sphinx for example.
+
+However, we believe that the initial effort to switch over to the new guidelines
+will be a worthwhile investment - in all cases it will mean being able to get
+rid of astropy-helpers as a submodule and the confusion and headaches this can
+cause, and it will make for example the definition of package data much simpler
+and not have to worry about ``setup_package.py`` files in most cases.
+
+The astropy `package-template`_ will be updated to reflect the latest
+recommendations, which will make it easier for developers to update their
+packages.
+
+Impact for contributors
+-----------------------
+
+Users who wish to contribute fixes to the core astropy package or other packages
+will be encouraged to have `tox`_ installed if they want to easily run
+tests or build documentation locally. However, this is an easy package to
+install with `pip`_ and we could also add code in ``setup.py`` so that running
+``python setup.py test`` or ``python setup.py build_docs`` gives a helpful
```

Comment:
Is it worth pointing out that we *can* potentially keep these commands, if they translate into `tox` or `pytest` commands?

Reply:
I added a note to this effect in the "alternatives" section.